File: 0686-repeated-string-match/0686-repeated-string-match.cpp
class Solution {
public:
    int repeatedStringMatch(string a, string b) {
        int n = a.size();
        int m = b.size();
        int count = 1;
        string str = a;
        
        while(str.size() < m) {
            str += a;
            count++;
        }
        
        if(str.find(b) != string::npos) {
            return count;
        }
        str += a;
        count++;
        
        if(str.find(b) != string::npos) {
            return count;
        }
        
        return -1;
    }
};

File: 0686-repeated-string-match/0686-repeated-string-match.java
class Solution {
    public int repeatedStringMatch(String a, String b) {
        String str = a;
        int count = 1;
        int m = b.length();

        while(str.length() < m) {
            str += a;
            count++;
        }

        if(str.indexOf(b) != -1) return count;

        str += a;
        count++;

        if(str.indexOf(b) != -1) return count;

        return -1;
    }
}